### PR TITLE
DSG-2233/rich text editor dom sanitizer

### PR DESCRIPTION
**Pull request by davidyoung77**
# Basics

## PR Type

What kind of change does this PR introduce?

- [ ] Bugfix
- [x] Feature
- [ ] Styling and layout changes
- [ ] Code style changes (formatting, local variables)
- [ ] Refactoring (no functional changes)
- [ ] Build or CI related changes
- [ ] Tests
- [ ] Other... Please describe:

## What changes does this PR introduce?

- Sanitize html for rich text editor
-- Preserve safe html
-- Remove unsafe html
- Enable wysiwyg alignment
- Enable wysiwyg color picker

## Screenshot of the affected changes

Screenshot 2023-11-07 at 12 28 49 PM
Screenshot 2023-11-07 at 12 29 25 PM

## Checklist

- [ ] I have updated the documentation accordingly
- [ ] I have confirmed all acceptance criteria defined by the story have been met
- [ ] I have confirmed that Sonar Qube findings for new code are reviewed and fixed, or will be whitelisted

### CODE REVIEW

File: libs/shared/ui/nuverial/src/lib/pipes/safe-html/safe-html.pipe.ts [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+export class NuverialSafeHtmlPipe implements PipeTransform {
+  constructor(private readonly _sanitizer: DomSanitizer) {}
+
+  public transform(htmlString?: string): SafeHtml {
+    const sanitizedContent = this.sanitizeHtml(htmlString);
+
+    return this._sanitizer.bypassSecurityTrustHtml(sanitizedContent);
+  }
+
+  public sanitizeHtml(htmlString?: string): string {
```
Comment (derekdiep42) [inline libs/shared/ui/nuverial/src/lib/pipes/safe-html/safe-html.pipe.ts:18]: nit: do we need this public? seems like it is only used here I think

**Review by derekdiep42 — APPROVED** [review]
Nice work, one small nit. LGTM!

**Review by jgnuvalence — APPROVED** [review]

File: libs/shared/ui/nuverial/src/lib/pipes/safe-html/safe-html.pipe.ts (hunk 1 above)
Comment (davidyoung77) [reply libs/shared/ui/nuverial/src/lib/pipes/safe-html/safe-html.pipe.ts:18]: It is public so that it can be used in a component

**Review by jgnuvalence — APPROVED** [review]

**Review by derekdiep42 — APPROVED** [review]
Looks good! Pending small nit.

**Review by derekdiep42 — APPROVED** [review]

**Pull request merged**